### PR TITLE
Update golang.org/x/net to v0.17.0

### PR DESCRIPTION
**Pull request by mborsz**
Necessary for: https://security-tracker.debian.org/tracker/CVE-2023-3978 and https://security-tracker.debian.org/tracker/CVE-2023-44487

### CODE REVIEW

**Comment by jkh52** [comment]
/ok-to-test

LGTM, but let's first cherry pick to `release-0.28` branch, before `release-0.1`.

**Comment by Argh4k** [comment]
Created https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/533 for release-0.28

**Comment by jkh52** [comment]
Did you see the test error? The other PR has it too.

```
	go get golang.org/x/net/idna@v0.17.0
../../../../pkg/mod/golang.org/x/net@v0.17.0/idna/idna10.0.0.go:27:2: missing go.sum entry for module providing package golang.org/x/text/unicode/bidi (imported by golang.org/x/net/idna); to add:
	go get golang.org/x/net/idna@v0.17.0
../../../../pkg/mod/golang.org/x/net@v0.17.0/idna/idna10.0.0.go:28:2: missing go.sum entry for module providing package golang.org/x/text/unicode/norm (imported by golang.org/x/net/idna); to add:
	go get golang.org/x/net/idna@v0.17.0
```

**Comment by jkh52** [comment]
Please `go mod tidy` here as well. The other one is merged so this is unblocked.

**Comment by mborsz** [comment]
I just pushed a version with go mod tidy. go test ./... works on my machine so it should be ok now.

File: konnectivity-client/go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ go 1.17
 require (
 	github.com/prometheus/client_golang v1.11.1
 	go.uber.org/goleak v1.2.0
-	golang.org/x/net v0.7.0 // indirect
+	golang.org/x/net v0.17.0 // indirect
```
Comment (jkh52) [inline konnectivity-client/go.mod:10]: client lib considerations:

Konnectivity `release-0.1` branch is used by k/k minor versions 1.26 and 1.27. I verified that https://github.com/kubernetes/kubernetes/pull/121117 has already been cherry picked to the relevant release branches there.

**Pull request merged**